### DISCUSSION TITLE
Memory-only / Testing database using MikroORM?

### DISCUSSION THREAD

**Discussion by Alwinator** [mikro-orm/mikro-orm#3056] [Q&A] [answered]
I am using MikroORM with SQLite for development and MikroORM with Postgres for production. However, for the tests Postgres would be overkill and SQLite would be a problem, especially when testing in parallel. I have e2e tests, so a mock would not be enough. What is the best way or database when testing with MikroORM?

**Comment by B4nan** [comment, marked as answer]
Using different driver for tests is something I would definitely not suggest, they will behave differently in edge cases, they don't support the same features, you will be testing things that can easily break in production (and you won't be able to test any postgres specific features). There is some in memory postgres emulator, still, it supports only a subset of features and can behave differently than a real database. If you want _end to end_ tests, you should use a real database.

**Reply by Alwinator** [reply]
Okay, thank you!

**Comment by darky** [comment]
I successfully using MikroORM SQLite in-memory for unit tests and MikroORM Postgres for e2e tests on my current job.
Once, I tired to run huge, time wasting Postgres e2e tests and switched to SQLite and almost all tests passed. Tiny 1-2 hour tuning and all my tests compatible with SQLite and Postgres at same time.
And no double work here, I write test once and run it twice, unit or e2e mode.
Everyday during all working day I run fast unit tests very often.
Rarely, before release of product, I run all this slow e2e tests.

**Comment by darky** [comment]
[Implementing of PGLite driver](https://github.com/mikro-orm/mikro-orm/discussions/5945) could totally resolve this problem in context of Postgres
